Objects/structseq.c

```diff
@@ -219,21 +218,21 @@ structseq_new_impl(PyTypeObject *type, PyObject *arg, PyObject *dict)
     }
     Py_DECREF(arg);
     for (; i < max_len; ++i) {
+        PyObject *ob;
         if (dict == NULL) {
-            ob = Py_None;
+            ob = Py_NewRef(Py_None);
         }
         else {
-            ob = _PyDict_GetItemStringWithError(dict,
-                type->tp_members[i-n_unnamed_fields].name);
+            const char *name = type->tp_members[i-n_unnamed_fields].name;
+            if (PyDict_GetItemStringRef(dict, name, &ob) < 0) {
+                Py_DECREF(res);
+                return NULL;
+            }
             if (ob == NULL) {
-                if (PyErr_Occurred()) {
-                    Py_DECREF(res);
-                    return NULL;
-                }
-                ob = Py_None;
+                ob = Py_NewRef(Py_None);
             }
         }
```

Comment:
It can be written more compactly, without duplication of `ob = Py_NewRef(Py_None)`.
```suggestion
        PyObject *ob = NULL;
        if (dict != NULL) {
            const char *name = type->tp_members[i-n_unnamed_fields].name;
                type->tp_members[i-n_unnamed_fields].name);
            if (PyDict_GetItemStringRef(dict, name, &ob) < 0) {
                Py_DECREF(res);
                return NULL;
            }
        }
        if (ob == NULL) {
            ob = Py_NewRef(Py_None);
        }
```